### PR TITLE
Remove superfluous datatype and device conversions

### PR DESCRIPTION
**Pull request by shadeMe**
## Description

Removes a couple of unnecessary type conversions and calls to `torch.to` when preparing inputs for the inner PyTorch transformer model.

### Types of change

Fix

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy_transformers/data_classes.py [hunk 1]
```diff
@@ -99,15 +99,17 @@ def from_batch_encoding(cls, token_data: BatchEncoding) -> "WordpieceBatch":
...
         ]
 
+        # The following tensors are intentionally allocated on the CPU to reduce
+        # host-to-device copies.
         numpy_ops = NumpyOps()
 
         return cls(
             strings=token_data["input_texts"],
-            input_ids=numpy_ops.asarray2i(token_data["input_ids"]),
+            input_ids=numpy_ops.asarray(token_data["input_ids"], dtype="int64"),
```
Comment (adrianeboyd) [inline spacy_transformers/data_classes.py:108]: I would refer to the dtype rather than hardcoding it? (Is it actually `int64`?)

Comment (shadeMe) [reply spacy_transformers/data_classes.py:108]: Good point. It is indeed `int64`, but I'll pass the source's `dtype` to `asarray`.

File: spacy_transformers/layers/transformer_model.py [hunk 2]
```diff
@@ -224,14 +224,11 @@ def _convert_transformer_inputs(model, wps: WordpieceBatch, is_train):
 
     hf_device = model.shims[0]._hfmodel.transformer.device
     kwargs = {
-        # Note: remove conversion to long when PyTorch >= 1.8.0.
```
Comment (adrianeboyd) [inline spacy_transformers/layers/transformer_model.py:227]: But there's not a corresponding change to the torch requirements?

File: spacy_transformers/data_classes.py [hunk 3]
```diff
@@ -99,16 +99,24 @@ def from_batch_encoding(cls, token_data: BatchEncoding) -> "WordpieceBatch":
             for tokens in token_data["input_texts"]
         ]
 
+        # The following tensors are intentionally allocated on the CPU to reduce
+        # host-to-device copies.
         numpy_ops = NumpyOps()
+        input_ids = token_data["input_ids"]
+        input_ids_dtype = input_ids.dtype if hasattr(input_ids, "dtype") else None
```
Comment (adrianeboyd) [inline spacy_transformers/data_classes.py:106]: Is there any chance that this doesn't have `.dtype`?

Comment (shadeMe) [reply spacy_transformers/data_classes.py:106]: Yeah, we pass lists in some of the tests.

File: spacy_transformers/layers/transformer_model.py (hunk 2 above)
Comment (shadeMe) [reply spacy_transformers/layers/transformer_model.py:227]: Fixed.

File: spacy_transformers/data_classes.py (hunk 3 above)
Comment (adrianeboyd) [reply spacy_transformers/data_classes.py:106]: Is it just in the `Dummy` test stuff?

Comment (shadeMe) [reply spacy_transformers/data_classes.py:106]: Don't think so/not sure - the [failing tests](https://dev.azure.com/explosion-ai/Public/_build/results?buildId=23801&view=logs&jobId=a25e3be4-cea9-5bf9-7ae0-24f1fc94349b&j=a25e3be4-cea9-5bf9-7ae0-24f1fc94349b&t=98bd6853-46be-5a37-e0f4-c5ea95fba4d8) were in `test_pipeline_component.py`.

Comment (adrianeboyd) [reply spacy_transformers/data_classes.py:106]: I'm pretty sure in practice it's always something with `.dtype` and we could update `DummyTokenizer`.

Comment (shadeMe) [reply spacy_transformers/data_classes.py:106]: Ok, I'll do that instead.

**Pull request merged**